com/http.go

```diff
@@ -0,0 +1,90 @@
+package com
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"serial-data-decryptor/models"
+	"serial-data-decryptor/processor"
+	"serial-data-decryptor/utility"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func StartServer() {
+	fmt.Println("Starting the server")
+
+	ln, err := net.Listen("tcp", utility.GetEnvAsserted("INGRESS_HOST")+":"+utility.GetEnvAsserted("INGRESS_PORT"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Started the server: " + ln.Addr().String())
+
+	for {
+		conn, err := ln.Accept() // this blocks until connection or error
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		go handleMessages(conn) // a goroutine handles conn so that the loop can accept other connections
+	}
+}
+
+func handleMessages(conn net.Conn) {
+	dec := gob.NewDecoder(conn)
+	msg := models.Data{}
+	dec.Decode(&msg)
```

Comment:
we should check for errors here and probably also make sure `msg` is not empty afterwards. please refer to docs of Decode

Reply:
added validator

Reply:
we should probably make an exception for EOF. I'm not sure if it's really relevant in our case

Reply:
I don't see it's needed